libraries/redcore/layouts/searchtools/default.php

```diff
@@ -23,7 +23,7 @@
 	'orderFieldSelector'  => '#list_fullordering'
 );
 
-$data['options'] = array_unique(array_merge($customOptions, $data['options']));
+$data['options'] = array_replace($customOptions, $data['options']);
```

Comment:
1. It works both actualy and in thats context do the same
2. array_merge slower than array_replace on ~10%:  for 10k loops with the same values
3. in joomla core has many bugs too and it should rechecks anyway, thats why we use redCORE, joomla not a strong reference.
Anyway bug in PR relate with `array_unique()`, it is remove key when i try to set `someValue = false`.